File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
# Time Complexity: O(n)
# Space Complexity: O(1)
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        insert_idx = 1
        size = len(nums)
        for i in range(1, size):
            # If the next number is a unique number:
            if nums[i] != nums[i-1]:
                nums[insert_idx] = nums[i]
                insert_idx += 1
        return insert_idx

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        if nums.is_empty() {
            return 0;
        }
        let mut insert_idx = 1;
        for i in 1..nums.len() {
            if nums[i] != nums[i-1] {
                nums[insert_idx] = nums[i];
                insert_idx += 1;
            }
        }
        insert_idx as i32
    }
}